### PR TITLE
[supertest] add host method

### LINKED ISSUE #51213: [@types/supertest] Host Method Missing
I found that the Supertest Agent has a host(string) method which sets the Host Header of requests from that agent, for the purpose of testing a domain-limited webserver locally which is currently not in the types:
`request.agent(myApp.callback()).host("abc.de")`
will error: Property 'host' does not exist on type 'SuperAgentTest'.ts(2339)
Logging 
`(<any>request.agent(myApp.callback())).host.toString()`
prints
`function(host) {
  this._host = host;
  return this;
}`
I could locate this function in
https://github.com/visionmedia/supertest/blob/master/lib/agent.js#L45
Please add it to the types. Thanks!
I would have contributed myself, but I looked through how to contribute to DefinetlyTyped and it is too complex for me.

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @NicoZelaya @mxl @paplorinc @shreyjain1994 @zopf @BeeeQueue @lukaselmer @theQuazz @carnesen @ghostganz

### PR DESCRIPTION
**Pull request by martin-badin**
Closes: #51213

URL: https://github.com/visionmedia/supertest/blob/master/lib/agent.js#L45

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/supertest/supertest-tests.ts [hunk 1]
```diff
@@ -43,6 +43,9 @@ supertest.agent(app, {
 // agent has request methods
 supertest.agent(app).set({ host: 'google.com' });
 
+// agent has host methods
+supertest.agent(app).host('something.test');
```
Comment (peterblazejewicz) [inline types/supertest/supertest-tests.ts:47]:
```suggestion
supertest.agent(app).host('something.test'); // $ExpectType Test
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@martin-badin  thx!

**Comment by martin-badin** [comment]
HI, @varju @pietu could someone review it?

**Review by peterblazejewicz — APPROVED** [review]

**Review by andrewbranch — APPROVED** [review]

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**